pkg/codec/renamer.go

```diff
@@ -0,0 +1,76 @@
+package codec
+
+import (
+	"reflect"
+	"unicode"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/types"
+)
+
+var pkgPath = reflect.TypeOf(renamer{}).PkgPath()
+
+func NewRenamer(fields map[string]string) Modifier {
+	m := &renamer{
+		modifierBase: modifierBase[string]{
+			fields:           fields,
+			onToOffChainType: map[reflect.Type]reflect.Type{},
+			offToOnChainType: map[reflect.Type]reflect.Type{},
+		},
+	}
+	m.modifyFieldForInput = func(field *reflect.StructField, _, newName string) error {
+		field.Name = newName
+		if unicode.IsLower(rune(field.Name[0])) {
+			field.PkgPath = pkgPath
+		}
```

Comment:
IIUC pkgpath is a unique identifier for unexported fields as exported fields are identified just by their name? Can there be some runtime issue if two fields with same name get assigned same PkgPath (in this case codec package path)

Reply:
But in a scenario where we have 2 jobs running that drop same fields there would be two filed with same name and package path right?

Reply:
I can be extra safe and make the type use the PkgPath of the type that's being modified.  Then if there are other private fields it won't break.  I didn't see a use case for the codec generating private fields though.

Reply:
I am just worried that if that if this somehow happens it would be a nightmare to track down

Reply:
> I can be extra safe and make the type use the PkgPath of the type that's being modified.

Yeah, this is how I was thinking it should work... makes more sense to me this way, thanks.